packages/rlc-common/src/buildClient.ts

```diff
@@ -308,9 +322,16 @@ export function getClientFactoryBody(
   const apiKeyHeaderName = credentialKeyHeaderName
     ? `apiKeyHeaderName: options.credentials?.apiKeyHeaderName ?? "${credentialKeyHeaderName}",`
     : "";
+  const loggerOptions =
+    branded !== false
+      ? `,
+  loggingOptions: {
+    logger: options.loggingOptions?.logger ?? logger.info
+  }`
+      : "";
 
   const credentialsOptions =
-    scopes || apiKeyHeaderName
+    (scopes || apiKeyHeaderName) && addCredentials
```

Comment:
Offline dicussed with @joheredi We may prefer to remove the option `addCredentials` from TypeSpec generation because TypeSpec could express no credential defined.

I will have a separated pr for this.